crates/autopilot/src/domain/settlement/mod.rs

```diff
@@ -200,19 +270,31 @@ impl From<U256> for TradeFlags {
     }
 }
 
-#[derive(Debug, thiserror::Error)]
-pub enum EncodingError {
-    #[error("unable to decode settlement calldata: {0}")]
-    Decoding(#[from] web3::ethabi::Error),
-    #[error("unable to tokenize calldata into expected format: {0}")]
-    Tokenizing(#[from] ethcontract::tokens::Error),
-    #[error("unable to recover order uid: {0}")]
-    OrderUidRecover(#[from] tokenized::Error),
-}
+pub mod error {
+    use super::*;
+
+    #[derive(Debug, thiserror::Error)]
+    pub enum Encoding {
+        #[error("unable to decode settlement calldata: {0}")]
+        Decoding(#[from] web3::ethabi::Error),
+        #[error("unable to tokenize calldata into expected format: {0}")]
+        Tokenizing(#[from] ethcontract::tokens::Error),
+        #[error("unable to recover order uid: {0}")]
+        OrderUidRecover(#[from] tokenized::Error),
+    }
```

Comment:
Do we use this type for anything else than the blanket `from` implementations?

I don't think those `#[from]`s are generally very good to use (they are convenient but bad), since now whenever you run into a `tokenized::Error` it will auto converted into `OrderUidRecover` even if let's say you are trying to parse the `AuctionId` at the end of the calldata. At the risk of writing 3 more lines of code, I think each callsite should decide which error type to throw (instead of making `?` work everywhere)

Reply:
I believe this was refactor as part of https://github.com/cowprotocol/services/pull/2746, so `from` is removed for `OrderUidRecover` variant.